File: Match/WebContent/scripts/editprofile.js
/**
 * 
 */

function printEditError(error) {
	$("#edit_error").empty().text(error);
};

$(function() {
	// add a function to the submit event
	$("#edit_form").submit(function() {
		//alert("submit check!!!");
		$(this).ajaxSubmit({
			dataType : 'json',
			error : function() {
				console.log("Failed to submit");
				printEditError("Failed to submit, please try one more time");
			},
			success : function(data) {
				
			
					
				
				if (data == "true") {
					$.mobile.changePage("#page4");
				} else {
					printEditError(data);
				}
			
			}
		});

		return false;
	});
});

$(document).on('pagebeforeshow', "#page5", function(event) {
	// on page load...do
	$.ajax({
		dataType : 'json',
		url : "profiledata",
		timeout : 2000,
		error : function() {
			console.log("Failed to submit");
		},
		success : function(data) {
			
			
			if(data == "go to sign in"){
				$.mobile.changePage("#page1");
			}
			else{
				
			
			refreshEditUserData(data);
			}
		}
	});

	$("#page5").trigger("create");
	return false;
});

function refreshEditUserData(userdatalist) {

	$("#update_first").attr("value", userdatalist.firstName);
	$("#update_last").attr("value", userdatalist.lastName);
	$("#update_about_me").empty().text(userdatalist.aboutMe);
	$("#update_sex").val(userdatalist.sex);
	$("#update_sex").selectmenu('refresh');
	$("#update_birthday").val(
			(Date.parse(userdatalist.birthday)).toString('yyyy-MM-dd'));
	$("#update_education").val(userdatalist.education);
	$("#update_education").selectmenu('refresh');
	$("#update_area").val(userdatalist.area);
	$("#update_area").selectmenu('refresh');
	$("#update_language").val(userdatalist.language);
	$("#update_language").selectmenu('refresh');
	$("#update_marital_status").val(userdatalist.maritalStatus);
	$("#update_marital_status").selectmenu('refresh');
	$("#update_children").val(userdatalist.children);
	$("#update_children").selectmenu('refresh');
	$("#update_eye_color").val(userdatalist.eyeColor);
	$("#update_eye_color").selectmenu('refresh');
	$("#update_hair_type").val(userdatalist.hairType);
	$("#update_hair_type").selectmenu('refresh');
	$("#update_hair_color").val(userdatalist.hairColor);
	$("#update_hair_color").selectmenu('refresh');
	$("#update_origin").val(userdatalist.origin);
	$("#update_origin").selectmenu('refresh');
	$("#update_height").val(userdatalist.height);
	$("#update_height").selectmenu('refresh');
	$("#update_weight").val(userdatalist.weight);
	$("#update_weight").selectmenu('refresh');
	$("#update_earning_level").val(userdatalist.earningLevel);
	$("#update_earning_level").selectmenu('refresh');
	$("#update_religion").val(userdatalist.religion);
	$("#update_religion").selectmenu('refresh');
	$("#update_smoking").val(userdatalist.smoking);
	$("#update_smoking").selectmenu('refresh');
	$("#update_physique").val(userdatalist.physique);
	$("#update_physique").selectmenu('refresh');
	$("#update_intrested_in").val(userdatalist.intrestedIn);
	$("#update_intrested_in").selectmenu('refresh');
	$("#update_photo").attr("src", userdatalist.photo);
	// ------------------Partner's Preferences----------------------------
	$("#update_education2").val(userdatalist.partner_education);
	$("#update_education2").selectmenu('refresh');
	$("#update_area2").val(userdatalist.partner_area);
	$("#update_area2").selectmenu('refresh');
	$("#update_language2").val(userdatalist.partner_language);
	$("#update_language2").selectmenu('refresh');
	$("#update_marital_status2").val(userdatalist.partner_maritalStatus);
	$("#update_marital_status2").selectmenu('refresh');
	$("#update_eye_color2").val(userdatalist.partner_eyeColor);
	$("#update_eye_color2").selectmenu('refresh');
	$("#update_hair_type2").val(userdatalist.partner_hairType);
	$("#update_hair_type2").selectmenu('refresh');
	$("#update_hair_color2").val(userdatalist.partner_hairColor);
	$("#update_hair_color2").selectmenu('refresh');
	$("#update_origin2").val(userdatalist.partner_origin);
	$("#update_origin2").selectmenu('refresh');
	$("#update_minheight2").val(userdatalist.partnerMinHeight);
	$("#update_minheight2").selectmenu('refresh');
	$("#update_maxheight2").val(userdatalist.partnerMaxHeight);
	$("#update_maxheight2").selectmenu('refresh');
	$("#update_earning_level2").val(userdatalist.partner_earningLevel);
	$("#update_earning_level2").selectmenu('refresh');
	$("#update_religion2").val(userdatalist.partner_religion);
	$("#update_religion2").selectmenu('refresh');
	$("#update_smoking2").val(userdatalist.partner_smoking);
	$("#update_smoking2").selectmenu('refresh');
	$("#update_physique2").val(userdatalist.partner_physique);
	$("#update_physique2").selectmenu('refresh');
	$("#update_min_age2").val(userdatalist.partnerMinAge);
	$("#update_min_age2").selectmenu('refresh');
	$("#update_max_age2").val(userdatalist.partnerMaxAge);
	$("#update_max_age2").selectmenu('refresh');

};

File: Match/src/actions/EditProfile.java
package actions;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.PrintWriter;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.logging.Level;
import java.util.logging.Logger;

import javax.servlet.ServletException;
import javax.servlet.annotation.MultipartConfig;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.Part;

import utilsAndData.JDBC;
import utilsAndData.SessionUtils;

import com.google.gson.Gson;

/**
 * Servlet implementation class EditProfile
 */
@WebServlet("/editprofile")
@MultipartConfig(location = "", fileSizeThreshold = 1024 * 1024, maxFileSize = 1024 * 1024 * 5, maxRequestSize = 1024 * 1024 * 5 * 5)
public class EditProfile extends HttpServlet {
	private static final long serialVersionUID = 1L;
	private JDBC jdbc = null;

	public EditProfile() {
		super();
	}
	
	public void init() throws ServletException {
		Object temp = getServletContext().getAttribute("matchDataJDBCTemplate");
		if (temp != null) {
			jdbc = (JDBC) temp;
		} else {
			throw new ServletException(
					"matchDataJDBCTemplate is not initiated.");
		}
	}

	protected void doGet(HttpServletRequest request,
			HttpServletResponse response) throws ServletException, IOException {
		try {
			processRequest(request, response);
		} catch (Exception ex) {
			Logger.getLogger(EditProfile.class.getName()).log(Level.SEVERE,
					null, ex);
		}
	}

	protected void doPost(HttpServletRequest request,
			HttpServletResponse response) throws ServletException, IOException {
		try {
			processRequest(request, response);
		} catch (Exception ex) {
			Logger.getLogger(EditProfile.class.getName()).log(Level.SEVERE,
					null, ex);
		}
	}

	protected synchronized void processRequest(HttpServletRequest request,
			HttpServletResponse response) throws ServletException, IOException,
			Exception {
		PrintWriter out = response.getWriter();
		String userIdFromSession = SessionUtils.getUserId(request);
		long userId = 0;
		Gson gson = new Gson();

		if (userIdFromSession == null) {
			String responce = "go to sign in";

			try {
				System.out.println("redirecting to sign in page");
				String jsonResponse = gson.toJson(responce);
				System.out.println(jsonResponse);
				out.print(jsonResponse);
				out.flush();
			} finally {
				out.close();
			}
		}

		else if (userIdFromSession != null) {

			try {
				userId = Long.parseLong(userIdFromSession);
			} catch (Exception e) {
				System.out.println("null session parameter");
			}

			String responce = "";
			getAndEditUserParameters(request, response, responce);

			try {
				Part filePart = request.getPart("file");
				if (filePart.getSize() != 0) {
					String fileType = filePart.getContentType();
					if (fileType.substring(0, 5).equalsIgnoreCase("image")) {
						System.out.println("type: " + fileType);
						
						InputStream inputStream = null;
						OutputStream outputStream = null;
						File img = new File("photo");
					 
						try {
							// read this file into InputStream
							inputStream = filePart.getInputStream();
							// write the inputStream to a FileOutputStream
							outputStream = new FileOutputStream(img);
					 
							int read = 0;
							byte[] bytes = new byte[1024];
					 
							while ((read = inputStream.read(bytes)) != -1) {
								outputStream.write(bytes, 0, read);
							}
					 
							System.out.println("Done!");
					 
						} catch (IOException e) {
							e.printStackTrace();
						} finally {
							if (inputStream != null) {
								try {
									inputStream.close();
								} catch (IOException e) {
									e.printStackTrace();
								}
							}
							if (outputStream != null) {
								try {
									outputStream.close();
								} catch (IOException e) {
									e.printStackTrace();
								}
					 
							}
						}

						jdbc.updateImages(img, (int)userId);
						responce = "true";
					} else {
						responce = "The type of image is not valid. Please choose jpg or png type of photo";
					}
				}else {
						responce = "true";
				}
			} catch (Exception ex) {
				responce = "Server failed to upload photo, please try again";
				System.out.println(ex.getLocalizedMessage());
				throw new IOException(ex);
			}

			try {
				String jsonResponse = gson.toJson(responce);
				System.out.println(jsonResponse);
				out.print(jsonResponse);
				out.flush();
			} finally {
				out.close();
			}
		}
	}

	private void getAndEditUserParameters(final HttpServletRequest request,
			final HttpServletResponse response, final String responce) {

		long userId = Long.parseLong(SessionUtils.getUserId(request));
		jdbc.updateUser(userId, "privacy_level", request.getParameter("privacy_level"));

		jdbc.updateUser(userId, "first", request.getParameter("update_first"));
		jdbc.updateUser(userId, "last", request.getParameter("update_last"));
		jdbc.updateUser(userId, "about_me",
				request.getParameter("update_about_me"));
		jdbc.updateUser(userId, "birthday",
				request.getParameter("update_birthday"));
		jdbc.updateUser(userId, "sex", request.getParameter("update_sex"));
		jdbc.updateUser(userId, "education",
				request.getParameter("update_education"));
		jdbc.updateUser(userId, "area", request.getParameter("update_area"));
		jdbc.updateUser(userId, "language",
				request.getParameter("update_language"));
		jdbc.updateUser(userId, "marital_status",
				request.getParameter("update_marital_status"));
		jdbc.updateUser(userId, "children",
				request.getParameter("update_children"));
		jdbc.updateUser(userId, "eye_color",
				request.getParameter("update_eye_color"));
		jdbc.updateUser(userId, "hair_type",
				request.getParameter("update_hair_type"));
		jdbc.updateUser(userId, "hair_color",
				request.getParameter("update_hair_color"));
		jdbc.updateUser(userId, "origin", request.getParameter("update_origin"));
		jdbc.updateUser(userId, "height", request.getParameter("update_height"));
		jdbc.updateUser(userId, "weight", request.getParameter("update_weight"));
		jdbc.updateUser(userId, "earning_level",
				request.getParameter("update_earning_level"));
		jdbc.updateUser(userId, "religion",
				request.getParameter("update_religion"));
		jdbc.updateUser(userId, "smoking",
				request.getParameter("update_smoking"));
		jdbc.updateUser(userId, "physique",
				request.getParameter("update_physique"));
		jdbc.updateUser(userId, "intrested_in",
				request.getParameter("update_intrested_in"));

		DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
		// get current date time with Date()
		Date date = new Date();
		jdbc.updateUser(userId, "last_login", dateFormat.format(date));

		jdbc.updateUser(userId, "partner_education",
				request.getParameter("update_education2"));
		jdbc.updateUser(userId, "partner_area",
				request.getParameter("update_area2"));
		jdbc.updateUser(userId, "partner_language",
				request.getParameter("update_language2"));
		jdbc.updateUser(userId, "partner_marital_status",
				request.getParameter("update_marital_status2"));
		jdbc.updateUser(userId, "partner_eye_color",
				request.getParameter("update_eye_color2"));
		jdbc.updateUser(userId, "partner_hair_type",
				request.getParameter("update_hair_type2"));
		jdbc.updateUser(userId, "partner_hair_color",
				request.getParameter("update_hair_color2"));
		jdbc.updateUser(userId, "partner_origin",
				request.getParameter("update_origin2"));
		jdbc.updateUser(userId, "partner_height_min",
				request.getParameter("update_minheight2"));
		jdbc.updateUser(userId, "partner_height_max",
				request.getParameter("update_maxheight2"));
		jdbc.updateUser(userId, "partner_religion",
				request.getParameter("update_religion2"));
		jdbc.updateUser(userId, "partner_smoking",
				request.getParameter("update_smoking2"));
		jdbc.updateUser(userId, "partner_earning_level",
				request.getParameter("update_earning_level2"));
		jdbc.updateUser(userId, "partner_physique",
				request.getParameter("update_physique2"));
		jdbc.updateUser(userId, "partner_age_min",
				request.getParameter("update_min_age2"));
		jdbc.updateUser(userId, "partner_age_max",
				request.getParameter("update_max_age2"));
	}
}
